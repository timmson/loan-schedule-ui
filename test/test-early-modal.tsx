import React from "react"

import {render, screen} from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import "regenerator-runtime"

import EarlyModal from "../src/early-modal"

const okButton = "ok"
const closeButton = "close"

describe("EarlyModal should", () => {


	test("ok by clicking 'Ok' button", async () => {
		render(
			<EarlyModal name={"name"} show={true} ok={{name: okButton, action: () => expect(true).toBeTruthy()}}
						close={{name: closeButton, action: () => expect(false).toBeTruthy()}}>
				X
			</EarlyModal>
		)

		await userEvent.click(screen.getByText(okButton))

		expect.assertions(1)
	})


	test("close by clicking 'Close' button", async () => {
		render(
			<EarlyModal name={"name"} show={true} ok={{name: "ok", action: () => expect(false).toBeTruthy()}} close={{name: closeButton, action: () => expect(true).toBeTruthy()}}>
				X
			</EarlyModal>
		)

		await userEvent.click(screen.getByText(closeButton))

		expect.assertions(1)
	})

	test("close by clicking 'cross'", async () => {
		render(
			<EarlyModal name={"name"} show={true} ok={{name: "ok", action: () => expect(false).toBeTruthy()}} close={{name: closeButton, action: () => expect(true).toBeTruthy()}}>
				X
			</EarlyModal>
		)

		await userEvent.click(screen.getByRole("button", {name: "Close"}))

		expect.assertions(1)
	})

})
