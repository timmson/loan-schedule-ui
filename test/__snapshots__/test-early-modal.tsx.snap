// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarlyModal should equal to snapshot 1`] = `null`;
